src/dash/models/DashManifestModel.js

```diff
@@ -292,6 +292,24 @@ function DashManifestModel(config) {
         return adaptation[DashConstants.CENC_DEFAULT_KID];
     }
 
+    function getLabel(adaptation) {
+        if (adaptation.lang && !adaptation.Label_asArray) {
+            return adaptation.lang;
+        }
+
+        if (adaptation.lang && adaptation.Label_asArray) {
+            var foundMatchingLabel = adaptation.Label_asArray.find((label) => label.lang === adaptation.lang);
```

Comment:
I don't understand this logic. I don't see why we someone could define more than one label for an adaptation if the one chosen is static and depends just on the lang property of the adaptation. 

@squarebracket, what do you think?

Reply:
I'm not sure I understand the spec completely, but I presume one could interpret it more than 1 way.

The way I looked at it was, even if the `lang` of the AdaptationSet was set to `en` one could define multiple labels (even if unnecessary) that look like this:

```
<AdaptationSet lang="en">
  <Label lang="fr">anglais</Label>
  <Label lang="en">english</Label>
  <Label lang="es">inglés</Label>
</AdaptationSet>
```

In such a case, we would want to `find` the correct label in the list of labels.

Reply:
I also assume that the intent of multiple `Label` elements is to display the label in several languages. That _seems_ to be the intent based on the screenshot that @sandersaares posted in #IOP on slack.

As for the selection logic, I think that's better left to those who know the dash.js code better. I'm not sure if or how localization/translation is done in dash.js, but I imagine that generally the device/browser locale would be used to select the display language. If that is done elsewhere in dash.js, then I think it would be reasonable to do it here. If not, then I think selecting the label based on the language of the AdaptationSet is reasonable -- an English one be shown as "English", a French one would be shown as "français", etc...

Reply:
IMHO, label selection depending on localization shouldn't be part of dash.js core. Seems like a logic that should be implemented at the "app level". What sounds better to me is dash.js exposing all the labels defined for a given adaptation and put the localization logic in controlbar.js. 

What do you think?

Reply:
Regarding Adaptation lang property, I think its purpose is declaring the language for the content of the AdaptationSet. Looks like we shouldn't assume we have to use a Russian label to describe a Russian audio track.

Reply:
The logic of separating out localization certainly tracks for me. I guess, then, the label field of the `TextTrackInfo` (and thus, the underlying native text track if it's used) would be empty?

I suppose since the `label` field doesn't present any pertinent information as it stands, it should be fine. We could use a separate property like `labels` which is just an array of label data, and then the control bar could use that.

That works for me. @sevagsim does that work for you?

Reply:
@epiclabsDASH @squarebracket - Agreed.  Working on an update now.

Reply:
@squarebracket @epiclabsDASH - Updated!